backend/src/main/java/gov/cdc/usds/simplereport/api/queue/QueueMutationResolver.java

```diff
@@ -34,22 +36,27 @@ public QueueMutationResolver(TestOrderService tos, PersonService ps) {
       _ps = ps;
     }
 
-    public void addTestResult(String deviceID, String result, String patientID, String dateTested) {
+    public void addTestResult(String deviceID, String result, String patientID, String dateTested) throws ParseException {
+      DateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+      Date isoDate = sdf.parse(dateTested);
       _tos.addTestResult(
         deviceID,
         TestResult.valueOf(result),
         patientID,
-        parseUserDate(dateTested) == null ? null : Date.from(parseUserDate(dateTested).atStartOfDay(ZoneId.systemDefault()).toInstant())
+        isoDate
       );
     }
 
-    public ApiTestOrder editQueueItem(String id, String deviceId, String result, String dateTested) {
-        return new ApiTestOrder(_tos.editQueueItem(
-            id,
-            deviceId,
-            result,
-            parseUserDate(dateTested) == null ? null : Date.from(parseUserDate(dateTested).atStartOfDay(ZoneId.systemDefault()).toInstant())
-        ));
+    public ApiTestOrder editQueueItem(String id, String deviceId, String result, String dateTested) throws ParseException {
```

Comment:
Same point about the input being a `Date` object rather than a string that we parse manually.